### ISSUE TITLE
aft-mtp-mount is not built in revision 4e16a11

### ISSUE THREAD

**Issue by sleepingscorpion** [whoozle/android-file-transfer-linux#114]
Upon successful build of revision '4e16a11', seems aft-mtp-mount binary executable is not built - ie does not exist in build output.

Attached it textfile showing output of 'ls -R' run from build dir.  

Result of 'ls -R | grep aft' is:

$ ls -R | grep aft
aft-mtp-cli
aft-mtp-cli.dir
./cli/CMakeFiles/aft-mtp-cli.dir:

Thanks
[ls-tree-output.txt](https://github.com/whoozle/android-file-transfer-linux/files/63125/ls-tree-output.txt)

**Comment by whoozle** [comment]
Yes, this is known bug, sorry, fuse was temporarily disabled due the API differences between MacOSX/Linux API. Please consider using aft-mtp-cli or android-file-transfer Qt UI. 

I hope @gazay will soon handle this issue.

**Comment by sleepingscorpion** [comment]
Ok, thanks for the quick response, I did search issues repo...but maybe not thoroughly enough! Cheers

**Comment by whoozle** [comment]
@sleepingscorpion may I ask you if it works for you? Does aft-mtp-cli or qt/android-file-transfer recognize the device? I have almost no MacOSX feeback, because there's no dmg or binary distribution yet.

**Comment by sleepingscorpion** [comment]
@whoozle, can connect on both, i9505 running CM12.1 rooted with USB debugging, in MTP mode. will test on LG V500 G-Pad tablet a little later...

Im happy to participate and contribute feedback on use cases/test cases anytime, just let me know what you need :)

aft-mtp-cli output:

$ ./cli/aft-mtp-cli
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Findno interface descriptor found
samsung GT-I9505 1.0
extensions: microsoft.com: 1.0; android.com: 1.0;
samsung GT-I9505> 

$ ./qt/android-file-transfer 
upload worker started 
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Find(*_interface)->USBInterfaceOpen(_interface): error 0xe00002c5
Device::Findno interface descriptor found
device found, opening session... 
device info "samsung"   "GT-I9505" 
switching to storage id  65537 
session opened, starting

**Comment by whoozle** [comment]
It's enough for now, thank you very much. 
Just to be sure that AFT works on MacOX for more than one computer/xcode/brew/device combination :)

**Comment by yurikoles** [comment]
@whoozle is there a flag to explicitly enable FUSE support?

I had created a formula for brew and explicitly add osxfuse dependency, which should setup env for supporting in, this how brew works, it builds a formula in sandbox. As far as I can see there is not checks for fuse in build log.

**Comment by whoozle** [comment]
@yurikoles yes, the option is still there for mac osx, it's just disabled by default. 
you can either pass -DBUILD_FUSE=TRUE to cmake, or edit CMakeCache.txt as follows
```
//Build fuse mount helper
BUILD_FUSE:BOOL=ON
```

**Comment by yurikoles** [comment]
When I try to build it manually I see this
```
[ 98%] Linking CXX executable aft-mtp-mount
ld: library not found for -losxfuse
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [fuse/aft-mtp-mount] Error 1
make[1]: *** [fuse/CMakeFiles/aft-mtp-mount.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
2 warnings generated.
[100%] Linking CXX executable aft-mtp-cli
[100%] Built target aft-mtp-cli
make: *** [all] Error 2
```

But it's here, under `/usr/local/lib`, and my updated brew formula builds and works well.

```$ ls /tmp/tissot/
'Внутрішнє спільне сховище'
```

**Comment by whoozle** [comment]
Awesome! It means that osxfuse api matches linux fuse, it's just library it could not find. 

Can you find it yourself? E.g. something like ```find /usr/local/lib -iname '*fuse*.so*' ```

**Comment by yurikoles** [comment]
```
$ find /usr/local/lib -name *fuse*
/usr/local/lib/pkgconfig/macfuse.pc
/usr/local/lib/pkgconfig/osxfuse.pc
/usr/local/lib/pkgconfig/fuse.pc
/usr/local/lib/libosxfuse_i64.dylib
/usr/local/lib/libosxfuse.2.dylib
/usr/local/lib/libfuse.dylib
/usr/local/lib/libfuse.la
/usr/local/lib/libosxfuse_i64.2.dylib
/usr/local/lib/libfuse.0.dylib
/usr/local/lib/libfuse.2.dylib
/usr/local/lib/libfuse_ino64.la
/usr/local/lib/libfuse_ino64.dylib
/usr/local/lib/libosxfuse.dylib
/usr/local/lib/libosxfuse_i64.la
/usr/local/lib/libosxfuse.la
/usr/local/lib/libfuse_ino64.2.dylib
```

**Comment by whoozle** [comment]
Nice! It looks like all libraries are in place. 
Can you open CMakeCache.txt again and look for the FUSE section, similar to the following:
```
FUSE_CFLAGS:INTERNAL=-I/usr/include/fuse;-D_FILE_OFFSET_BITS=64
FUSE_CFLAGS_I:INTERNAL=
FUSE_CFLAGS_OTHER:INTERNAL=-D_FILE_OFFSET_BITS=64
FUSE_FOUND:INTERNAL=1
FUSE_INCLUDEDIR:INTERNAL=/usr/include
FUSE_INCLUDE_DIRS:INTERNAL=/usr/include/fuse
FUSE_LDFLAGS:INTERNAL=-lfuse;-pthread
FUSE_LDFLAGS_OTHER:INTERNAL=-pthread
FUSE_LIBDIR:INTERNAL=/usr/lib64
FUSE_LIBRARIES:INTERNAL=fuse
```

Is there any cmake var with something like -L/usr/local/lib or just  /usr/local/lib in its value?

**Comment by yurikoles** [comment]
Yes, it's there: `FUSE_STATIC_LDFLAGS:INTERNAL=-L/usr/local/lib;-losxfuse`,
In full:
```
FUSE_CFLAGS:INTERNAL=-D_FILE_OFFSET_BITS=64;-D_DARWIN_USE_64_BIT_INODE;-I/usr/local/include/osxfuse/fuse
FUSE_CFLAGS_I:INTERNAL=
FUSE_CFLAGS_OTHER:INTERNAL=-D_FILE_OFFSET_BITS=64;-D_DARWIN_USE_64_BIT_INODE
FUSE_FOUND:INTERNAL=1
FUSE_INCLUDEDIR:INTERNAL=/usr/local/include
FUSE_INCLUDE_DIRS:INTERNAL=/usr/local/include/osxfuse/fuse
FUSE_LDFLAGS:INTERNAL=-L/usr/local/lib;-losxfuse
FUSE_LDFLAGS_OTHER:INTERNAL=
FUSE_LIBDIR:INTERNAL=/usr/local/lib
FUSE_LIBRARIES:INTERNAL=osxfuse
FUSE_LIBRARY_DIRS:INTERNAL=/usr/local/lib
FUSE_LIBS:INTERNAL=
FUSE_LIBS_L:INTERNAL=
FUSE_LIBS_OTHER:INTERNAL=
FUSE_LIBS_PATHS:INTERNAL=
FUSE_PREFIX:INTERNAL=/usr/local
FUSE_STATIC_CFLAGS:INTERNAL=-D_FILE_OFFSET_BITS=64;-D_DARWIN_USE_64_BIT_INODE;-I/usr/local/include/osxfuse/fuse
FUSE_STATIC_CFLAGS_I:INTERNAL=
FUSE_STATIC_CFLAGS_OTHER:INTERNAL=-D_FILE_OFFSET_BITS=64;-D_DARWIN_USE_64_BIT_INODE
FUSE_STATIC_INCLUDE_DIRS:INTERNAL=/usr/local/include/osxfuse/fuse
FUSE_STATIC_LDFLAGS:INTERNAL=-L/usr/local/lib;-losxfuse
FUSE_STATIC_LDFLAGS_OTHER:INTERNAL=
FUSE_STATIC_LIBDIR:INTERNAL=
FUSE_STATIC_LIBRARIES:INTERNAL=osxfuse
FUSE_STATIC_LIBRARY_DIRS:INTERNAL=/usr/local/lib
FUSE_STATIC_LIBS:INTERNAL=
FUSE_STATIC_LIBS_L:INTERNAL=
FUSE_STATIC_LIBS_OTHER:INTERNAL=
FUSE_STATIC_LIBS_PATHS:INTERNAL=
FUSE_VERSION:INTERNAL=2.9.7
FUSE_fuse_INCLUDEDIR:INTERNAL=
FUSE_fuse_LIBDIR:INTERNAL=
FUSE_fuse_PREFIX:INTERNAL=
FUSE_fuse_VERSION:INTERNAL=
```

**Comment by yurikoles** [comment]
I remind that today new cmake is shipped in Homebrew, maybe it's some new bug.

**Comment by whoozle** [comment]
nope, it's definitely caused by missing path in aft cmakelists.
could you pull it and try to build it again? 
9c222187bd1a86499252e2aaad662d1d2fcb83c9

**Comment by yurikoles** [comment]
It works now, I'm going to add it to dmg. Don't you mind isse a new tag after it?

**Comment by yurikoles** [comment]
It's fixed now.
